File: src/java/FibonacciRecursive.java
public class FibonacciRecursive {

  public static void main(String[] args) {
    System.out.println("Fib(16) : " + FibonacciRecursive(16));
  }

  public static int FibonacciRecursive(int x) {
    if (x <= 1) return 1;
    else return FibonacciRecursive(x - 1) + FibonacciRecursive(x - 2);
  }
}

File: src/scala/FibonacciRecursive.scala
object FibonacciRecursive {
  def fibonacciRecursive(n: Int): BigInt = {
    if (n <= 1) {
      return n
    }

    fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2)
  }

  def main(args: Array[String]): Unit = {
    val index = 15
    println(s"Fibonacci (recursive) of $index is: ${fibonacciRecursive(index)}")
  }
}
